subsys/usb/usb_c/usbc_prl.c

```diff
@@ -392,7 +394,9 @@ static void prl_hr_set_state(const struct device *dev, const enum usbc_prl_hr_st
 	struct usbc_port_data *data = dev->data;
 	struct protocol_hard_reset_t *prl_hr = data->prl_hr;
 
+#if !defined(CONFIG_TEST)
```

Comment:
Why remove the assert when `CONFIG_TEST` is enabled? The point behind asserts is to alert programmer about conditions that must never be false but do become false due to incorrect code (e.g. invalid API usage).

Reply:
I agree that this is not the correct thing to do but twister was complaining:
```
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/util.h:109:45: error: invalid application of 'sizeof' to incomplete type 'const struct smf_state[]'
  109 |         ((size_t) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
      |                                             ^
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/__assert.h:116:23: note: in definition of macro '__ASSERT'
  116 |                 if (!(test)) {                                            \
      |                       ^~~~
/home/sam/Projects/github/zephyrproject/zephyr/subsys/usb/usb_c/usbc_prl.c:395:26: note: in expansion of macro 'ARRAY_SIZE'
  395 |         __ASSERT(state < ARRAY_SIZE(prl_hr_states), "invalid prl_hr_state %d", state);
```
But the build works fine.